client/src/shared/state/Inventory.js

```diff
@@ -104,16 +110,76 @@ class Inventory {
         if (!ItemTypes[itemConfig.typeCode].hasUseEffect) return;
 
         this.hotbar.push(itemConfig);
-
+        this.saveItemId[itemConfig.id] = itemConfig.slotIndex;
         PubSub.publish(HOTBAR_ITEM);
         PubSub.publish(MODIFY_INVENTORY_ITEM);
+        if (this.saveSession === true) {
+            this.saveHotbar();
+        }
     }
 
     removeFromHotbar(itemConfig) {
+        delete this.saveItemId[itemConfig.id];
         this.hotbar = this.hotbar.filter((eachItem) => eachItem !== itemConfig);
-
         PubSub.publish(HOTBAR_ITEM);
         PubSub.publish(MODIFY_INVENTORY_ITEM);
+        if (this.saveSession === true) {
+            this.saveHotbar();
+        }
+    }
+
+    saveHotbar() {
+        window.localStorage.setItem(this.loadHotBarRequest, JSON.stringify(this.saveItemId));
+    }
+
+    defaultHotBar() {
+        this.items.forEach((itemConfig) => {
+            this.addToHotbar(itemConfig);
+        });
+    }
+
+    loadFromLocalStorage(playerID) {
+        this.saveSession = true;
+        this.userName = playerID;
+        const gameName = "dungeonz";
+        const requestHotbar = "hotBar";
+        this.loadHotBarRequest = gameName + this.userName + requestHotbar;
+    }
+
+    loadHotbar(playerID, isLoggedIn) {
+        if (isLoggedIn === true) {
+            this.loadFromLocalStorage(playerID);
+        }
+        const loadStorageHotbar = window.localStorage.getItem(this.loadHotBarRequest);
+        if (this.saveSession === false || loadStorageHotbar === null) {
+            this.defaultHotBar();
+            return;
+        }
+        this.saveItemId = JSON.parse(loadStorageHotbar);
+        this.populateInvetorytoHotbar();
+    }
+
+    populateInvetorytoHotbar() {
+        Object.keys(this.saveItemId).forEach((key, value) => {
+            this.items.forEach((itemConfig) => {
+                if (key === itemConfig.id) {
+                    this.addToHotbar(itemConfig);
+                }
+            });
+        });
+
+        // For the case there is a mismatch in session remove non-existing keys
+        Object.keys(this.saveItemId).forEach((key, value) => {
+            this.foundKey = false;
+            this.hotbar.forEach((itemConfig) => {
+                if (itemConfig.id === key) {
+                    this.foundKey = true;
+                }
+            });
+            if (this.foundKey === false) {
+                delete this.saveItemId[key];
```

Comment:
Will the local storage need to be updated again after any of these invalid ones are found?

Reply:
I am thinking this may not be needed at all. Say that initially there was a mismatch, the item would not be added to the hotbar because the ID is not found, it will initially only load the items, that still exists in inventory. Then, say they add or remove an item from hotbar, the hotbar will automatically be able to adjust itself, so this extra loop is kind of redundant because we can guarantee if a mismatch happens, and then in the new session, after adding or deleting an item, the hashmap would automatically be able to adjust itself. Opinion?

Reply:
See my comment https://github.com/Arcanorum/dungeonz/pull/216/files#r767337810 about why this may not be needed.